src/snowflake/snowpark/mock/plan.py

```diff
@@ -895,6 +909,215 @@ def calculate_expression(
         column = calculate_expression(exp.child, input_data, analyzer, expr_to_alias)
         column.sf_type = ColumnType(exp.to, exp.nullable)
         return column
+    if isinstance(exp, WindowExpression):
+        # Should return one column
+        window_function = exp.window_function
+        window_spec = exp.window_spec
+
+        # Process order by clause
+        if window_spec.order_spec:
+            res = input_data.sort_values(
+                by=[
+                    exp.child.name for exp in window_spec.order_spec
+                ],  # TODO: consider when the expr is something like col1+2
+                ascending=[
+                    isinstance(exp.direction, Ascending)
+                    for exp in window_spec.order_spec
+                ]
+                or True,
+            )
+        else:
+            res = input_data
+
+        res_index = res.index  # List of row indexes of the result
+
+        # Process partition_by clause
+        if window_spec.partition_spec:
+            res = res.groupby(
+                [exp.name for exp in window_spec.partition_spec],
+                sort=False,
+                as_index=False,
+            )
+            res_index = []
+            for r in res:
+                res_index += list(r[1].index)
+
+        if not window_spec.frame_spec or not isinstance(
+            window_spec.frame_spec, SpecifiedWindowFrame
```

Comment:
can remind me what else classes could this `window_spec.frame_spec` be?
from the logic it looks we only support `SpecifiedWindowFrame`,  if there are other `window_spec.frame_spec` that we don't support we should probably document it in the limitations

Reply:
there is `UnspecifiedFrame` which means to use the default window frame, pleas see https://docs.snowflake.com/en/sql-reference/functions-analytic#window-frame-usage-notes.